### PR TITLE
Use KMSKeyState when creating and deleting keys

### PR DESCRIPTION
**Pull request by rossf7**
Fixes problems with creating and deleting keys that I missed in my previous PR.

I've also changed the resource methods to use the KMSKeyState struct rather than the AWS SDK structs. I think this is simpler and matches what we do for the S3 bucket.

### CODE REVIEW

File: service/resource/kmskeyv2/delete.go [hunk 1]
```diff
@@ -10,24 +10,27 @@ import (
...
 	}
 
-	keyAlias := deleteAliasInput.AliasName
-	if keyAlias != nil && *keyAlias != "" {
-		if _, err := r.awsClients.KMS.DeleteAlias(&deleteAliasInput); err != nil {
-			return microerror.Mask(err)
-		}
-
+	if deleteInput.KeyAlias != "" {
+		// Get the KMS key ID using the key alias.
```
Comment (rossf7) [inline service/resource/kmskeyv2/delete.go]: Now we get the key Arn using the alias before deleting it. So we can be sure we have the right value.

File: service/resource/kmskeyv2/create.go [hunk 2]
```diff
@@ -50,19 +42,19 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
...
 
-	desiredKMSState, err := toKMSKeyState(desiredState)
+	desiredKeyState, err := toKMSKeyState(desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	if desiredKMSState.KeyAlias != currentKMSState.KeyAlias {
-		return &kms.CreateKeyInput{}, nil
+	if currentKeyState.KeyAlias == "" || desiredKeyState.KeyAlias != currentKeyState.KeyAlias {
```
Comment (rossf7) [inline service/resource/kmskeyv2/create.go]: Conditional now matches the s3ucket resource and I think is more robust.

File: service/resource/kmskeyv2/delete.go [hunk 3]
```diff
@@ -57,22 +60,22 @@ func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desire
...
+		return nil, microerror.Mask(err)
 	}
 
 	r.logger.LogCtx(ctx, "debug", "finding out if the KMS key should be deleted")
 
-	if currentKeyState.KeyAlias != "" && desiredKeyState.KeyAlias != currentKeyState.KeyAlias {
-		deleteChange.AliasName = aws.String(currentKeyState.KeyAlias)
+	var kmsKeyToDelete KMSKeyState
+	if currentKeyState.KeyAlias != "" {
+		kmsKeyToDelete = desiredKeyState
```
Comment (rossf7) [inline service/resource/kmskeyv2/delete.go]: Delete check also updated to be consistent with the s3bucket resource.

File: service/resource/kmskeyv2/create.go [hunk 4]
```diff
@@ -6,30 +6,22 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/kms"
 	"github.com/giantswarm/microerror"
-
-	"github.com/giantswarm/aws-operator/service/keyv2"
 )
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
-	createKeyInput, err := toCreateKeyInput(createChange)
+	createInput, err := toKMSKeyState(createChange)
```
Comment (fgimenez) [inline service/resource/kmskeyv2/create.go]: :+1:

**Pull request merged**